demosplan/DemosPlanCoreBundle/Event/Procedure/PostProcedureUpdatedEvent.php

```diff
@@ -12,21 +12,66 @@
 
 namespace demosplan\DemosPlanCoreBundle\Event\Procedure;
 
+use DemosEurope\DemosplanAddon\Contracts\Entities\EntityInterface;
 use DemosEurope\DemosplanAddon\Contracts\Events\PostProcedureUpdatedEventInterface;
 use demosplan\DemosPlanCoreBundle\Entity\Procedure\Procedure;
 use demosplan\DemosPlanCoreBundle\Event\DPlanEvent;
+use ReflectionClass;
 
 class PostProcedureUpdatedEvent extends DPlanEvent implements PostProcedureUpdatedEventInterface
 {
-    protected Procedure $procedure;
+    protected Procedure $procedureBeforeUpdate;
+    protected Procedure $procedureAfterUpdate;
 
-    public function __construct(Procedure $procedure)
+    public function __construct(Procedure $procedureBeforeUpdate, Procedure $procedureAfterUpdate)
     {
-        $this->procedure = $procedure;
+        $this->procedureBeforeUpdate = $procedureBeforeUpdate;
+        $this->procedureAfterUpdate = $procedureAfterUpdate;
     }
```

Comment:
Can be simplified via [constructor properties](https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion). You can combine this with the usage of [`readonly` properties](https://www.php.net/manual/en/language.oop5.properties.php).

Reply:
adjusted in [3412c03](https://github.com/demos-europe/demosplan-core/pull/1741/commits/3412c03af57cc1035ebac82fe51bb21d79ef2e59)